packages/@ourworldindata/utils/src/Util.ts

```diff
@@ -1291,6 +1292,25 @@ export const getArticleFromJSON = (
     }
 }
 
+export function recursivelyMapArticleBlock(
+    block: OwidEnrichedArticleBlock,
+    callback: (block: OwidEnrichedArticleBlock) => OwidEnrichedArticleBlock
+): OwidEnrichedArticleBlock {
+    if (block.type === "gray-section") {
+        block.items.map((block) => recursivelyMapArticleBlock(block, callback))
```

Comment:
`forEach` since we're not using the result of the `map`? (another instance below) might not be relevant if refactoring into the other traversal function as mentioned above.